### DISCUSSION TITLE
Model Convergence and Low Baseline

### DISCUSSION THREAD

**Discussion by shubhambooking2** [google/meridian#196] [General]
Iam using daily level data of last three years and have ~20 media channels and control variable as google search query volume data.

i have been exploring the below approaches:

- using Default Beta Priors + knots at monthly/weekly level : However here the problem being faced is the baseline is very low and sometimes coming to be negative. I have not given a strong prior and also i have controls like google search volume data , but still this issue which is making me wonder what possibly could be the reason

- Using Default ROI Priors : While using ROI priors my model finishes in ~3 mins, despite giving lager number of chains and n_adpat and looking at diagnostics, its clear that model failed to converge because of a huge Rhat value (in order of thousands). This makes me wonder what could be the reason that in case of ROI prior model completely fails to converge whereas in case of Beta as prior thats not the case

- Interesting the convergence is not the issue when using weekly data with ROI priors.

would be great to know some insights or suggestions for the above three points. Thanks

**Comment by shubhambooking2** [comment]
Its indeed confusing - i have tried multiple iterations with different combinations (including/excluding knots, controls) but in all cases the daily level model with ROI prior is giving a very huge R hat(in order of thousands). Where as this is not the case when i use the weekly model

Also Daily level model works perfectly fine when i use Beta Priors.  So now iam quite confused why with only ROI priors for daily model it fails, even though when we pass ROI prior those are tranlated to Beta internally. Any Suggestions/Inputs would be helpful!

**Comment by jyklly** [comment]
It looks like you're already assessing convergence and trying out solutions but will just point out [https://developers.google.com/meridian/docs/advanced-modeling/additional-considerations#getting-mcmc-convergence](https://developers.google.com/meridian/docs/advanced-modeling/additional-considerations#getting-mcmc-convergence) in case there are additional things there to try. 

It may be the case that the aggregation to weekly from daily is helping to smooth out the data allowing Meridian to identify the signal in the data and making it easier to converge. 

The default beta prior and default ROI prior are not directly equivalent and hence can regulate the inference in different ways. So it's possible to get convergence using beta priors for the daily data but not ROI. 

Have you tried any non-default ROI priors in daily? Assuming a more informative prior may help convergence here as the default may be unrealistic for the data. Could also look at setting the total media contribution prior as mentioned in
[https://github.com/google/meridian/discussions/188#discussioncomment-9907992](https://github.com/google/meridian/discussions/188#discussioncomment-9907992).

**Comment by shubhambooking2** [comment]
I guess i missed out pointing this but as you rightly suggested in https://developers.google.com/meridian/docs/advanced-modeling/additional-considerations#getting-mcmc-convergence i have tried most of the suggestions mentioned here for example removing correlated variables, trying out informative prior, changing altogether the default lognormal to other  distribution such as Weibull, Uniform etc and have tried the option of total media contribution prior option, but in all of these scenarios the issue is pretty much the same.
